license-tool/src/main.rs

```diff
@@ -0,0 +1,313 @@
+#![deny(rust_2018_idioms)]
+#![warn(clippy::pedantic)]
+#![allow(clippy::redundant_closure_for_method_calls)]
+
+use anyhow::{bail, Context, Result};
+use argh::FromArgs;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fs::{self, File};
+use std::io;
+use std::path::{Path, PathBuf};
+use url::Url;
+
+/// Stores arguments
+#[derive(FromArgs, PartialEq, Debug)]
+struct Args {
+    /// configuration file with the licenses to be used
+    #[argh(option, short = 'l')]
+    licenses_file: String,
+
+    #[argh(subcommand)]
+    subcommand: Subcommand,
+}
+
+/// Stores the subcommand to be executed
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand)]
+enum Subcommand {
+    SpdxId(SpdxIdArgs),
+    Path(PathArgs),
+    Fetch(FetchArgs),
+}
+
+/// Returns the spdx-id for the package
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand, name = "spdx-id")]
+struct SpdxIdArgs {
+    /// the package name used to look up for the licenses
+    #[argh(option, short = 'p')]
+    package_name: String,
+}
+
+/// Creates a copy of the licenses files in the dest directory
+#[derive(FromArgs, Debug, PartialEq)]
+#[argh(subcommand, name = "fetch")]
+struct FetchArgs {
+    /// the destination folder for the licenses
+    #[argh(option, short = 'd')]
+    destination: PathBuf,
```

Comment:
Same nit here - if we only take one option we could just make it a positional argument.